### PR TITLE
MenuSearch - fix keyboard shortcuts, Gatsby example improvement

### PR DESCRIPTION
**Pull request by laraharrow**
✨ Changes

- Fixes a bug where keyboard shortcuts were only working with tab but not arrows.
- Gatsby example improvement - implement the use of a usable search

✅ Requirements
   - [ ] Includes test coverage for all changes
   - [x] Build and tests are passing
   - [x] PR is ideally < 400LOC
   - [ ] Link(s) to related Github issues

📷 Screenshots

- Search is working for Getsby example

correct-behavior

- Keyboard arrow working to focus on all elements listed
keyboard-arrow

### CODE REVIEW

**Comment by mdodgelooker** [comment]
Oh also can you add in the formatted description and checklist similar to https://github.com/looker-open-source/components/pull/33 ?

File: packages/www/src/documentation/components/overlays/menu.mdx [hunk 1]
```diff
@@ -253,6 +253,50 @@ It supports an optional `label` prop to place a heading above the `MenuItem`s. W
...
+  const handleChange = event => setKeywordSearch(event.currentTarget.value)
+
+  const handleClear = () => setKeywordSearch('')
+
+  React.useEffect(() => {
+    const results = defaultCheeses.filter(cheese =>
+      cheese.toLowerCase().includes(keywordSearch.toLowerCase())
+    )
+    setSearchResults(results)
+  }, [keywordSearch])
```
Comment (mdodgelooker) [inline packages/www/src/documentation/components/overlays/menu.mdx]: Remove this whole section and replace with:
```
searchResults = defaultCheeses.filter(cheese =>
  cheese.toLowerCase().includes(keywordSearch.toLowerCase())
)
```

File: packages/www/src/documentation/components/overlays/menu.mdx [hunk 2]
```diff
@@ -253,6 +253,50 @@ It supports an optional `label` prop to place a heading above the `MenuItem`s. W
 ```jsx
 ;() => {
   const menuRef = React.useRef(null)
+  const defaultCheeses = ['Gouda', 'Swiss', 'Cheddar', 'Goat', 'Parmesan']
+
+  const [keywordSearch, setKeywordSearch] = React.useState('')
+  const [searchResults, setSearchResults] = React.useState(defaultCheeses)
```
Comment (mdodgelooker) [inline packages/www/src/documentation/components/overlays/menu.mdx]: Remove this line.

**Comment by lukebowerman** [comment]
Please squash this when merging :)

**Pull request merged**